### DISCUSSION TITLE
snowflake table training

### DISCUSSION THREAD

**Discussion by tmohanta** [vanna-ai/vanna#422] [Q&A]
**Describe the bug**
Using the fllowing  code to training the snowflake table. But traing is not effective
Hi All, I am trying train my snowflake  tables  but RESULTS  are not good. Using ollama3,sqldb as vectordb

training_plan = vn.get_training_plan_snowflake(
            filter_databases=[MY_SNOWFLAKE_DATABASE],  # Optional: Filter databases
            filter_schemas=[MY_SNOWFLAKE_SCHEMA],       # Optional: Filter schemas
            include_information_schema=False, # Optional: Include system schema
            use_historical_queries=False       # Optional: Include historical queries
        )
        print(training_plan)
        # Filter the plan
        filtered_plan = []
        for item in training_plan._plan:
            if item.item_group != '' and item.item_name in (MY_TABLE):
                filtered_plan.append(item)
        training_plan._plan = filtered_plan
        # 1.Train the model using the generated training plan
        print("----plan---" ,training_plan)
        vn.train(plan=training_plan)
The ddl train as follows
id = vn.train(ddl="""CREATE TABLE IF NOT EXISTS  MY_CBOT.TESTS.MY_TABLE (
                TEST_DETAILS_ID NUMBER(38,0),
                TEST_ID VARCHAR(50),
                TEST_CLT_ID NUMBER(38,0),
                TEST_CLT_NAME VARCHAR(255),
                TEST_STATUS VARCHAR(255),MY_JSON VARIANT
                )"""
        )
**To Reproduce**

**Expected behavior**

**Error logs/Screenshots**
If applicable, add logs/screenshots to give more information about the issue.

**Desktop (please complete the following information where):**
 - OS: [e.g. Ubuntu]
 - Version: [e.g. 20.04]
 - Python: [3.9]
 - Vanna: [2.8.0]

**Additional context**

**Comment by zainhoda** [comment]
Please provide more information: "But traing is not effective" and "but RESULTS are not good" is not particularly helpful.
